### PR TITLE
CorrDiff: Enhance configs, remove custom `module.py`

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request
closes https://github.com/NVIDIA/modulus/issues/367 and https://github.com/NVIDIA/modulus/issues/311

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: examples/generative/corrdiff/train.py [hunk 1]
```diff
@@ -57,10 +58,9 @@ def main(cfg: DictConfig) -> None:
     OmegaConf.save(cfg, os.path.join(cfg.outdir, "config.yaml"))
 
     # Parse options
-    regression_checkpoint_path = getattr(cfg, "regression_checkpoint_path", None)
+    regression_checkpoint_path = getattr(cfg, to_absolute_path("regression_checkpoint_path"), None)
```
Comment (akshaysubr) [inline examples/generative/corrdiff/train.py:61]: Shouldn't this be
```python
regression_checkpoint_path = getattr(cfg, "regression_checkpoint_path", None)
if regression_checkpoint_path:
    regression_checkpoint_path = to_absolute_path(regression_checkpoint_path)
```

File: examples/generative/corrdiff/train.py [hunk 2]
```diff
@@ -57,10 +58,9 @@ def main(cfg: DictConfig) -> None:
...
     task = getattr(cfg, "task")
-    outdir = getattr(cfg, "outdir", "./output")
-    data = getattr(cfg, "data", "./data")
+    outdir = getattr(cfg, to_absolute_path("outdir"), "./output"))
```
Comment (akshaysubr) [inline examples/generative/corrdiff/train.py:63]: Same comment as above

File: examples/generative/corrdiff/train.py (hunk 1 above)
Comment (akshaysubr) [reply examples/generative/corrdiff/train.py:61]: Similarly for all the other paths in the lines below

**Review by akshaysubr — CHANGES_REQUESTED** [review]
Looks good to me. Just flagging one issue with the config paths that might be a mistake.

**Review by dallasfoster — APPROVED** [review]
Approved pending @akshaysubr comments.

File: examples/generative/corrdiff/train.py (hunk 1 above)
Comment (mnabian) [reply examples/generative/corrdiff/train.py]: Fixed

File: examples/generative/corrdiff/train.py (hunk 2 above)
Comment (mnabian) [reply examples/generative/corrdiff/train.py]: Fixed

File: examples/generative/corrdiff/docs/2023-arxiv/figures/scores_for_table.py
Comment (dallasfoster) [inline examples/generative/corrdiff/docs/2023-arxiv/figures/scores_for_table.py:1]: I really don't feel like these hard-coded paths are appropriate in a public repository.

Comment (akshaysubr) [reply examples/generative/corrdiff/docs/2023-arxiv/figures/scores_for_table.py:1]: Agreed, there are quite a few hard-coded paths. @mnabian Can we remove them from here or at least create an issue to remove them in a follow up PR?

Comment (mnabian) [reply examples/generative/corrdiff/docs/2023-arxiv/figures/scores_for_table.py:1]: https://github.com/NVIDIA/modulus/issues/393

**Pull request merged**